example-app/src/main/java/com/adyen/checkout/example/data/api/CheckoutApiService.kt

```diff
@@ -39,26 +37,24 @@ internal interface CheckoutApiService {
     @POST("paymentMethods")
     suspend fun paymentMethodsAsync(@Body paymentMethodsRequest: PaymentMethodsRequest): PaymentMethodsApiResponse
 
-    // There is no native support for JSONObject in either Moshi or Gson, so using RequestBody as a work around for now
     @POST("payments")
-    fun payments(@Body paymentsRequest: PaymentsRequest): Call<ResponseBody>
+    fun payments(@Body paymentsRequest: JSONObject): Call<JSONObject>
 
-    // There is no native support for JSONObject in either Moshi or Gson, so using RequestBody as a work around for now
     @POST("payments")
-    suspend fun paymentsAsync(@Body paymentsRequest: PaymentsRequest): ResponseBody
+    suspend fun paymentsAsync(@Body paymentsRequest: JSONObject): JSONObject
 
     @POST("payments/details")
-    fun details(@Body detailsRequest: JSONObject): Call<ResponseBody>
+    fun details(@Body detailsRequest: JSONObject): Call<JSONObject>
 
     @POST("payments/details")
-    suspend fun detailsAsync(@Body detailsRequest: JSONObject): ResponseBody
+    suspend fun detailsAsync(@Body detailsRequest: JSONObject): JSONObject
```

Comment:
Is it okay to return `JSONObject`s or should we use data classes?

Reply:
`JSONObject` is good enough for the example app 👍  I think it's even better if it "just works" instead of using proper classes since we don't want anyone to replicate this behavior and make API calls form the client app.